### ISSUE TITLE
`GetListItems` modifies input `fields` list

### ISSUE THREAD

**Issue by amnonkhen** [jasonrollins/shareplum#44]
Calling `GetListItems` with the `fields` parameter as a list modifies the input, while calling it with a tuple fails the method.

Here is the [culprit snippet](https://github.com/jasonrollins/shareplum/blob/master/shareplum/shareplum.py#L451):

```python
    # Add viewFields
    if fields:
        # Convert to SharePoint Style Column Names
        for i, val in enumerate(fields):
            fields[i] = self._disp_cols[val]['name']
```
 
as a workaround the input fields lists can be cloned.

**Comment by jasonrollins** [comment]
GetListItems 100% modifies the fields.  Is there a reason you need to maintain the original?

**Comment by DA-Mena** [comment]
@amnonkhen I agree it's kinda weird that one of my arguments when passed in as another variable... it gets modified and you're forced to have have inline a new list object upon every GetListItems call.

**Comment by jasonrollins** [comment]
This is a design decision.  I can change it if it causes real world problems.  I'm closing this issue for now.

**Comment by jasonrollins** [comment]
I changed my mind about leaving this as-is.  The default is now to not change the data.  This is fixed in v0.4.3.
